socketioxide/src/packet.rs

```diff
@@ -338,7 +338,7 @@ impl TryFrom<String> for Packet {
 
         let data = chars.as_str();
         let inner = match index {
-            '0' => PacketData::Connect(deserialize_packet(data)?.unwrap_or(json!({}))),
+            '0' => PacketData::Connect(deserialize_packet(data)?.unwrap_or_else(|| json!({}))),
```

Comment:
I doesn't seem to change anything ?

Reply:
True, it's a clippy warning though so I just applied its suggestion.

Reply:
So, I did some more research and apparantly `unwrap_or_else` is faster in this case:

> The primary reason to use ok_or_else or any of the ..._or_else methods is to avoid executing a function when it's not needed. In the case of Option::ok_or_else or Option::unwrap_or_else, there's no need to run extra code when the Option is Some. This can make code faster, depending on what happens in the error case

Clippy assumes the given function may produce side effects and thus should not be evaluated immediately. The general recommendation seems to be to use `unwrap_or` for constant values and `unwrap_or_else` for functions. 

Although the `json!` macro doesn't currently induce such side effects, I like this rule of thumb so I'm considering keeping it as is. Unless you disagree, of course.